File: src/main/java/GHAI/ai/SurveyService.java
package GHAI.ai;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.*;

@Service
public class SurveyService {

    @Autowired
    private QuestionRepository questionRepository;

    @Autowired
    private ResponseRepository responseRepository;

    // Get general questions
    public List<Question> getGeneralQuestions() {
        return questionRepository.findBySurveyType("GENERAL");
    }

    // Get follow-up questions based on responses
    public List<Question> getFollowUpQuestions(List<Response> responses) {
        Set<String> detectedCategories = new HashSet<>();

        for (Response response : responses) {
            Question question = questionRepository.findById(response.getQuestionId()).orElse(null);
            if (question != null && question.getCategory() != null) {
                detectedCategories.add(question.getCategory());
            }
        }

        if (detectedCategories.isEmpty()) {
            return Collections.emptyList();
        }

        return questionRepository.findByCategoryIn(new ArrayList<>(detectedCategories));
    }

    // Analyze responses to determine probable diagnosis
    public String analyzeResponses(List<Response> responses) {
        Map<String, Integer> categoryCount = new HashMap<>();

        for (Response response : responses) {
            Question question = questionRepository.findById(response.getQuestionId()).orElse(null);
            if (question != null && question.getCategory() != null) {
                categoryCount.put(question.getCategory(), categoryCount.getOrDefault(question.getCategory(), 0) + 1);
            }
        }

        return categoryCount.entrySet().stream()
                .max(Map.Entry.comparingByValue())
                .map(Map.Entry::getKey)
                .orElse("Uncertain - More Tests Needed");
    }

    // Get all questions
    public List<Question> getAllQuestions() {
        return questionRepository.findAll();
    }

    // Get questions by survey type
    public List<Question> getQuestionsBySurveyType(String surveyType) {
        return questionRepository.findBySurveyType(surveyType);
    }

    // Get questions by category
    public List<Question> getQuestionsByCategory(String category) {
        return questionRepository.findByCategory(category);
    }
}

File: client/src/services/SurveyService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/survey'; // Adjust the URL based on your backend

export const getGeneralQuestions = () => {
    return axios.get(`${API_URL}/questions/general`);
};

export const getFollowUpQuestions = (responses) => {
    return axios.post(`${API_URL}/questions/followup`, responses);
};

export const submitSurvey = (responses) => {
    return axios.post(`${API_URL}/submit`, responses);
};

export const getAllQuestions = () => {
    return axios.get(`${API_URL}/questions/all`);
};

export const getQuestionsBySurveyType = (surveyType) => {
    return axios.get(`${API_URL}/questions`, { params: { surveyType } });
};

export const getQuestionsByCondition = (condition) => {
    return axios.get(`${API_URL}/questions/condition`, { params: { condition } });
};